common/djangoapps/student/views.py

```diff
@@ -1599,6 +1611,39 @@ def create_account_with_params(request, params):
         getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {})
     )
 
+    # Start: Added by Labster
+    # User can create an account only in the appropriate region or user
+    # that already enrolled in this current region.
+    if not CourseEnrollmentAllowed.objects.filter(email=params["email"]).exists():
+        base_url = request.build_absolute_uri('/')[:-1]
+        regions = configuration_helpers.get_value('REGIONS', settings.REGIONS)
+        # Take out the url of this current region in the list of regions.
+        for region_code, values in regions.items():
+            if base_url in values['login_url']:
+                del regions[region_code]
+                break
+
+        # List of country codes: https://dev.maxmind.com/geoip/legacy/codes/iso3166/
+        current_region = request.session.get('country_code')
+        # If the user region is covered by Labster, suggest student to
+        # create an account to the appropriate region server.
+        region_info = regions.get(current_region)
+        if region_info:
+            scheme, netloc, path, query_string, fragment = urlsplit(region_info['login_url'])
+            register_url = urlunsplit((scheme, netloc, reverse("register_user"), '', ''))
```

Comment:
it's better to be consistent and this url as a parameter to the configuration.

Reply:
updated, Thanks